src/_pytest/config/__init__.py

```diff
@@ -998,6 +998,7 @@ def __init__(
         self.hook.pytest_addoption.call_historic(
             kwargs=dict(parser=self._parser, pluginmanager=self.pluginmanager)
         )
+        self.args_source = Config.ArgsSource.ARGS
```

Comment:
Any reason why not initializing `args` here as well?

Reply:
because of this
```python
assert not hasattr(
            self, "args"
        ), "can only parse cmdline args at most once per Config object"

Reply:
Ahh I see. Took the liberty of changing the assertion and move `args` to `__init__`. 👍